enterprise/frontend/src/embedding-sdk/lib/polyfill-use-sync-external-store.ts

```diff
@@ -1,11 +1,11 @@
 import React from "react";
 import { useSyncExternalStore } from "use-sync-external-store/shim";
 
-import { isReact17OrEarlier } from "metabase/lib/react-compat";
-
 // Monkey-patches useSyncExternalStore if we are in React 17,
 // where useSyncExternalStore is not available.
 // This is used in react-redux v8, until they've dropped the shim in v9.
-if (isReact17OrEarlier()) {
+export const shouldShimExternalStore = () => !("useSyncExternalStore" in React);
+
+if (shouldShimExternalStore()) {
```

Comment:
nit: the function is exported but not used anywhere, I think with the comment above it's clear what this is doing
```suggestion
if ("useSyncExternalStore" in React) {
```